src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -127,42 +127,48 @@ extension RiskProvider: RiskProviding {
 		userInitiated: Bool,
 		ignoreCachedSummary: Bool = false,
 		appConfiguration: SAP_ApplicationConfiguration,
-		completion: @escaping (SummaryMetadata?) -> Void
+		completion: @escaping (Result<SummaryMetadata, RiskCalculationError>) -> Void
 	) {
 		Log.info("RiskProvider: Determine summeries.", log: .riskDetection)
+
 		if !ignoreCachedSummary {
-			// Here we are in automatic mode and thus we have to check the validity of the current summary
+			func completeWithCachedSummary() {
```

Comment:
This is only called once, what's the reasoning behind extracting this into an inner func?

Reply:
removed inner func